plugins/spammer/plugin.go

```diff
@@ -48,8 +50,20 @@ func configure(plugin *node.Plugin) {
 		spammerWorkerCount = 1
 	}
 
+	if maxCPUUsage > 0.0 {
+		_, err := CPUUsage()
+		if err != nil {
+			maxCPUUsage = 0.0
+			log.Infof("Ignoring maxCPUUsage config setting. %s", err)
+		} else {
+			rateLimit = 0.0
+			spammerWorkerCount = runtime.NumCPU() - 1 // note: having more workers in this case actually seems to make the cpu usage more stable. Keeping the amount 'sane' for now though
+			// log.Infof("maxCPUUsage=%.2f rateLimit=%.2f spammerWorkerCount=%d", maxCPUUsage, rateLimit, spammerWorkerCount)
+		}
+	}
+
 	if rateLimit != 0 {
-		rateLimitChannelSize := int64(rateLimit) * 2
+		rateLimitChannelSize := int64(rateLimit) * 2 // XXX this case looks wrong and I doubt this code works as intended
```

Comment:
Why? It creates a channel with double the size of the rate limit. No bigger size needed.

Reply:
My comment should have been removed before checking in.

Do we need the rateLimit channel to be buffered?